sabnzbd/nzbstuff.py

```diff
@@ -1151,6 +1151,10 @@ def get_extra_blocks(self, setname: str, needed_blocks: int) -> int:
                 new_nzf = block_list.pop()
```

Comment:
We should probably just make this a `for`-loop and don't do the popping but just `break` when `added_blocks >= needed_blocks`?
Then we can do an `else` to the `for` that `return 0` in case it loops through all without ever getting enough blocks.